File: src/main/java/arcanitor/civilengineering/bridge/ApiMessage.java
package arcanitor.civilengineering.bridge;

import arcanitor.civilengineering.CivilEngineering;
import arcanitor.civilengineering.Config;
import com.google.gson.Gson;

public class ApiMessage {
    private String text = "";
    private String channel = "";
    private String username = "";
    private String userid = "";
    private String avatar = "";
    private String gateway = "";
    private String event = "";

    public ApiMessage(String user, String msg) {
        this.username = user;
        this.text = msg;
        this.gateway = Config.gateway;

    }
    public ApiMessage(String user, String msg, String event) {
        this.username = user;
        this.text = msg;
        this.event = event;
    }

    public static ApiMessage decode(String json) {
        Gson gson = new Gson();
        ApiMessage msg = gson.fromJson(json, ApiMessage.class);
        return msg;

    }

    public String encode() {
        Gson gson = new Gson();
        return gson.toJson(this);
    }

    public String getUsername() {
        return this.username;
    }

    public String getMessage() {
        return this.text;
    }

    public String getEvent() {
        return this.event;
    }



}

File: core/src/main/kotlin/matterlink/bridge/ApiMessage.kt
package matterlink.bridge

import matterlink.config.cfg
import matterlink.antiping
import com.google.gson.Gson
import matterlink.mapFormat

const val USER_ACTION: String = "user_action"
const val JOIN_LEAVE: String = "join_leave"

data class ApiMessage(
        val username: String = cfg.relay.systemUser,
        val text: String = "",
        val gateway: String = cfg.connect.gateway,
        val channel: String = "",
        val userid: String = "",
        val avatar: String = "",
        val account: String = "",
        val event: String = "",
        val protocol: String = "",
//        val timestamp: Date,
        val id: String = ""
//        val Extra: Any? = null
) {


    companion object {
        val gson = Gson()

        fun decode(json: String): ApiMessage {
            return gson.fromJson(json, ApiMessage::class.java)
        }
    }

    fun encode(): String {
        return gson.toJson(this)
    }

    fun format(fmt: String): String {
        return fmt.mapFormat(
                mapOf(
                        "{username}" to username,
                        "{text}" to text,
                        "{gateway}" to gateway,
                        "{channel}" to channel,
                        "{protocol}" to protocol,
                        "{username:antiping}" to username.antiping()
                )
        )

    }
}